### PR TITLE
feat(button-like-components): add helper function for enter clicking

### PR DESCRIPTION
**Pull request by LawrenceBorst**
Added helper function that you would put in componentWillLoad that adds the event listeners needed to make buttons clickable with enter.

We add it to limel-buttons to make testing easy (this will probably conflict with another similar issue, and we need to fix this conflict before merge).

Fixes Lundalogik/crm-feature#3518

## How to test this PR:
I tested it by throwing in a timeout in `componentWillLoad` that calls the add/remove event. I suggest trying something similar, calling them in different orders, maybe repeatedly, and see if it breaks.

The docs have a really good playground for the limel-button to test this functionality.

## Review:
- [ ] Commits are [atomic](https://seesparkbox.com/foundry/atomic_commits_with_git)
- [ ] Commits have the correct *type* for the changes made
- [ ] Commits with *breaking changes* are marked as such

### Browsers tested:
(Check any that applies, it's ok to leave boxes unchecked if testing something didn't seem relevant.)

Windows:
- [ ] Chrome
- [ ] Edge
- [ ] Firefox

Linux:
- [ ] Chrome
- [ ] Firefox

macOS:
- [ ] Chrome
- [ ] Firefox
- [ ] Safari

Mobile:
- [ ] Chrome on Android
- [ ] iOS

### CODE REVIEW

File: src/components/button/button.tsx [hunk 1]
```diff
@@ -66,8 +67,15 @@ export class Button {
     @State()
     private justLoaded = false;
 
+    @Element()
+    private host: HTMLElement;
+
     private justLoadedTimeout?: number;
 
+    public componentWillLoad() {
+        makeClickable(this.host);
```
Comment (LawrenceBorst) [inline src/components/button/button.tsx:76]: Hm come to think of it, this is a bad name 😅 
I'll probably change it to something like `makeEnterClickable`

File: src/util/makeClickable.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
+import { forceUpdate } from '@stencil/core';
+
+export function makeClickable(host) {
+    let isActive;
+    let hasJustReleasedEnter;
+
+    if (isActive === undefined || hasJustReleasedEnter === undefined) {
```
Comment (LawrenceBorst) [inline src/util/makeClickable.ts:7]: This we can take out. Just initialise it above

File: src/util/makeClickable.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+                hasJustReleasedEnter = false;
+
+                return;
+            }
+
+            event.stopImmediatePropagation();
+        },
+        true
+    );
+}
```
Comment (jgroth) [inline src/util/makeClickable.ts:46]: We should probably allow for some cleanup as well, if a consumer no longer wants the element to have this behavior. Now, once you have called this function you are stuck with the behavior forever.

File: src/components/button/button.tsx [hunk 4]
```diff
@@ -1,5 +1,5 @@
 import { Component, h, Prop, State, Watch, Element } from '@stencil/core';
-import { makeClickable } from '../../util/makeClickable';
+import { makeEnterClickable } from '../../util/makeClickable';
```
Comment (LucyChyzhova) [inline src/components/button/button.tsx:2]: don't forget to change the file name to `makeEnterClickable.ts`

**Review by LucyChyzhova — APPROVED** [review]
Tested, works well!  🚀

**Pull request merged**